File: golang/SRP/solution/employee/employee.go
package employee_solution

type Employee struct {
	employeeID         string
	employeeName       string
	employeeAddress    string
	contactNumber      string
	employeeType       string
	employeeRepository EmployeeRepository
	taxCalculator      TaxCalculator
}

func (e *Employee) Save() {
	e.employeeRepository.Save(e)
}
func (e *Employee) CalculateTax() {
	e.taxCalculator.CalculateTax(e)
}
func (e *Employee) GetEmployeeID() string {
	return e.employeeID
}
func (e *Employee) SetEmployeeID(ID string) {
	e.employeeID = ID
}
func (e *Employee) GetEmployeeName() string {
	return e.employeeName
}
func (e *Employee) SetEmployeeName(employeeName string) {
	e.employeeName = employeeName
}
func (e *Employee) GetEmployeeAddress() string {
	return e.employeeAddress
}
func (e *Employee) SetEmployeeAddress(employeeAddress string) {
	e.employeeAddress = employeeAddress
}
func (e *Employee) GetContactNumber() string {
	return e.contactNumber
}
func (e *Employee) SetContactNumber(contactNumber string) {
	e.contactNumber = contactNumber
}
func (e *Employee) GetEmployeeType() string {
	return e.employeeType
}
func (e *Employee) SetEmployeeType(employeeType string) {
	e.employeeType = employeeType
}

File: javalang/src/SRP/solution/Employee.java
package SRP.solution;

public class Employee {
	private String employeeID;         
	private String employeeName;
	private String employeeAddress;
	private String contactNumber;
	private String employeeType;

	public void save() {
		new EmployeeRepository().save(this);
	}
	public void calculateTax() {
		new TaxCalculator().calculateTax(this);
	}
	public String getEmployeeID() {
		return this.employeeID;
	}	
	public void setEmployeeID(String ID) {
		this.employeeID=ID;
	}
	public String getEmployeeName() {
		return this.employeeName;
	}
	public void setEmployeeName(String employeeName) {
		this.employeeName = employeeName;
	}
	public String getEmployeeAddress() {
		return this.employeeAddress;
	}
	public void setEmployeeAddress(String employeeAddress) {
		this.employeeAddress = employeeAddress;
	}
	public String getContactNumber() {
		return this.contactNumber;
	}
	public void setContactNumber(String contactNumber) {
		this.contactNumber = contactNumber;
	}
	public String getEmployeeType() {
		return this.employeeType;
	}
	public void setEmployeeType(String employeeType) {
		this.employeeType = employeeType;
	}
}